lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -822,95 +822,142 @@ computeBinary(ExprSolution lhs, ExprSolution rhs,
 /// to memoize the result of expressions in case they were not involved in a
 /// cycle (which may alter their value from the perspective of a variable).
 static ExprSolution solveExpr(Expr *expr, SmallPtrSetImpl<Expr *> &seenVars,
-                              unsigned indent = 1) {
-  // See if we have a memoized result we can return.
-  if (expr->solution) {
+                              unsigned defaultWorklistSize) {
+
+  struct Frame {
+    Expr *expr;
+    unsigned indent;
+  };
+
+  // indent only used for debug logs.
+  unsigned indent = 1;
+  std::vector<Frame> worklist({{expr, indent}});
```

Comment:
nit: Is there a specific reason to use std::vector over SmallVector?

Reply:
I don't think so, the worklist can grow pretty large, we can switch to smallvector, how does the logic for `reserve` look to you ? I reserved the number of exprs/2 size for this.

Reply:
I don't have good intuition for the constraint but exprs/2 sounds reasonable.